mlflow/projects/utils.py

```diff
@@ -0,0 +1,133 @@
+        raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(no_command_message))
+
+    if 'parameters' in entry_point_entries:
+        for parameter_name, parameter_yaml in entry_point_yaml.get('parameters').items():
+            _validate_entry_point_parameter_yaml(entry_point_name, parameter_name, parameter_yaml)
+
+
+def _validate_entry_point_parameter_yaml(entry_point_name, parameter_name, parameter_yaml):
+    """helper to validate parameter yaml within an entry point"""
+    parameter_allowed_entries = ('type', 'default')
+
+    # define this here so we can provide a more informative message
+    def check_parameter_type(attempted_type):
+        if attempted_type not in MLPROJECT_PARAMETER_TYPES:
+            unsupported_parameter_type_message = (
+                "unsupported type for parameter {} in entry point {}, "
+                "parameter value will be converted to string"
+            ).format(parameter_name, entry_point_name)
+            _logger.warning(unsupported_parameter_type_message)
+
+    # interpret the entry as a type if it's a single string
+    if isinstance(parameter_yaml, str):
+        check_parameter_type(parameter_yaml)
+        return
+
+    # otherwise expect sub-entries, at least for 'type', potentially also for 'default'
+    parameter_entries = parameter_yaml.keys()
+    _validate_entries_are_allowed('parameter', parameter_entries, parameter_allowed_entries)
+
+    if 'type' not in parameter_entries:
+        missing_parameter_type_message = "parameter {} in entry point {} must specify a type" \
+            .format(parameter_name, entry_point_name)
+        raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(missing_parameter_type_message))
+
+    # TODO: validate 'default' parameter value somehow
+    if 'default' in parameter_entries:
+        pass
+
+    check_parameter_type(parameter_yaml.get('type'))
+
+
+def _validate_docker_env_yaml(docker_env_yaml):
+
+    if not isinstance(docker_env_yaml, dict) or not isinstance(docker_env_yaml.get('image'), str):
+        bad_docker_entry_message = (
+            "docker_env must be a YAML object with a string 'image' entry representing "
+            "a Docker image that is accessible on the system executing the project"
+        )
+        raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(bad_docker_entry_message))
+
+    docker_env_entries = docker_env_yaml.keys()
+    _validate_entries_are_allowed('docker env', docker_env_entries, ('image',))
+
+
+def _validate_conda_env_yaml(conda_env_yaml):
+    # TODO: make this condition more specific
+    if not isinstance(conda_env_yaml, str):
+        bad_conda_env_message = (
+            "conda_env should be a string representing the relative path to a "
+            "Conda environment YAML file in the MLflow project's directory"
+        )
+        raise ExecutionException(BAD_MLPROJECT_MESSAGE.format(bad_conda_env_message))
+
+
+def _validate_name_yaml(name_yaml):
+    if not isinstance(name_yaml, str):
```

Comment:
```suggestion
    valid_types = six.string_types + [six.text_type]
    if not any([isinstance(name_yaml, valid_type) for valid_type in valid_types]) :
```

In offline discussion @dbczumar had mentioned that directly checking if the type is a `str` here could lead to throwing on unicode names - I think the suggested alternative avoids this issue